components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/SchemaValidator.java

```diff
@@ -138,15 +135,10 @@ public boolean handleResponse(MessageContext messageContext) {
         if (objectResponse == null) {
             return true;
         }
-        String contentType = objectResponse.toString();
-        JSONObject payloadObject = getMessageContent(messageContext);
-        if (APIMgtGatewayConstants.APPLICATION_JSON.equals(contentType) &&
-                payloadObject != null && !APIMgtGatewayConstants.EMPTY_ARRAY.equals(payloadObject)) {
-            try {
-                validateResponse(messageContext);
-            } catch (APIManagementException e) {
-                logger.error("Error occurred while validating the API response", e);
-            }
+        try {
+            validateResponse(messageContext);
+        } catch (APIManagementException e) {
+            logger.error("Error occurred while validating the API response", e);
```

Comment:
Same here sInce we just logging the error if there is any exception will validation get ommitted?

Reply:
Yes, I fixed the issue by setting return false.